bless/aws_lambda/bless_lambda_lyft_host.py

```diff
@@ -0,0 +1,303 @@
+    :param ca_private_key_password: For local testing, if the password is provided, skip the KMS
+    decrypt.
+    :param entropy_check: For local testing, if set to false, it will skip checking entropy and
+    won't try to fetch additional random from KMS
+    :param config_file: The config file to load the SSH CA private key from, and additional settings
+    :return: the SSH Certificate that can be written to id_rsa-cert.pub or similar file.
+    """
+    # AWS Region determines configs related to KMS
+    region = os.environ['AWS_REGION']
+
+    # Load the deployment config values
+    if config_file is None:
+        config_file = os.path.join(os.getcwd(), 'bless_deploy.cfg')
+    config = BlessLyftConfig(region, config_file=config_file)
+
+    certificate_type = get_certificate_type(config.get(BLESS_OPTIONS_SECTION, CERTIFICATE_TYPE_OPTION))
+    logging_level = config.get(BLESS_OPTIONS_SECTION, LOGGING_LEVEL_OPTION)
+    numeric_level = getattr(logging, logging_level.upper(), None)
+    if not isinstance(numeric_level, int):
+        raise ValueError('Invalid log level: {}'.format(logging_level))
+
+    logger.setLevel(numeric_level)
+
+    certificate_validity_window_seconds = config.getint(BLESS_OPTIONS_SECTION,
+                                                        CERTIFICATE_VALIDITY_WINDOW_SEC_OPTION)
+    entropy_minimum_bits = config.getint(BLESS_OPTIONS_SECTION, ENTROPY_MINIMUM_BITS_OPTION)
+    random_seed_bytes = config.getint(BLESS_OPTIONS_SECTION, RANDOM_SEED_BYTES_OPTION)
+    cross_account_role_arn = config.get(BLESS_OPTIONS_SECTION, CROSS_ACCOUNT_ROLE_ARN_OPTION)
+    ca_private_key_file = config.get(BLESS_CA_SECTION, CA_PRIVATE_KEY_FILE_OPTION)
+    password_ciphertext_b64 = config.getpassword()
+    kmsauth_key_id = config.get(BLESS_CA_SECTION, KMSAUTH_KEY_ID_OPTION)
+    kmsauth_context = config.get(BLESS_CA_SECTION, KMSAUTH_CONTEXT_OPTION)
+
+    # read the private key .pem
+    with open(os.path.join(os.getcwd(), ca_private_key_file), 'r') as f:
+        ca_private_key = bytes(f.read(), 'ascii')
+
+    # decrypt ca private key password
+    if ca_private_key_password is None:
+        kms_client = boto3.client('kms', region_name=region)
+        ca_password = kms_client.decrypt(
+            CiphertextBlob=base64.b64decode(password_ciphertext_b64))
+        ca_private_key_password = ca_password['Plaintext']
+
+    # if running as a Lambda, we can check the entropy pool and seed it with KMS if desired
+    if entropy_check:
+        with open('/proc/sys/kernel/random/entropy_avail', 'r') as f:
+            entropy = int(f.read())
+            logger.debug(entropy)
+            if entropy < entropy_minimum_bits:
+                logger.info(
+                    'System entropy was {}, which is lower than the entropy_'
+                    'minimum {}.  Using KMS to seed /dev/urandom'.format(
+                        entropy, entropy_minimum_bits))
+                response = kms_client.generate_random(
```

Comment:
I'm not entirely sure if we need to address right now, but I noticed if `entropy_check` is `True`, then `kms_client` will not be initialized unless `ca_private_key_password` is set (https://github.com/lyft/bless/pull/43/files#diff-a69a8ab6712afafbc98a768d892b3e56R206)

This seems to be the same as the old branch and seems to be operating okay.  But maybe a quick additional line:

```
if not kms_client:
    kms_client = boto3.client('kms', region_name=region)
```